File: ios/Sources/ExamplePlugin/ExamplePlugin.swift
import Foundation
import Capacitor

@objc(ExamplePlugin)
public class ExamplePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ExamplePlugin"
    public let jsName = "ExamplePlugin"
    private let implementation = Example()

    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "set", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "get", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "remove", returnType: CAPPluginReturnPromise)
    ]

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value", "")
        call.resolve(["value": implementation.echo(value)])
    }

    @objc func get(_ call: CAPPluginCall) {
        let key = call.getString("key", "")
        let value = implementation.get(key: key) ?? ""
        call.resolve(["value": value])
    }

    @objc func set(_ call: CAPPluginCall) {
        let key = call.getString("key", "")
        let value = call.getString("value", "")
        implementation.set(key: key, value: value)
        call.resolve()
    }

    @objc func remove(_ call: CAPPluginCall) {
        let key = call.getString("key", "")
        implementation.remove(key: key)
        call.resolve()
    }
}

File: android/src/main/java/com/mycompany/plugins/example/ExamplePlugin.java
package com.mycompany.plugins.example;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ExamplePlugin")
public class ExamplePlugin extends Plugin {

    private Example implementation;

    @Override
    public void load() {
        implementation = new Example(getContext());
    }

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value", "");
        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void set(PluginCall call) {
        String key = call.getString("key", "");
        String value = call.getString("value", "");
        implementation.set(key, value);
        call.resolve();
    }

    @PluginMethod
    public void get(PluginCall call) {
        String key = call.getString("key", "");
        String value = implementation.get(key);
        JSObject ret = new JSObject();
        ret.put("value", value);
        call.resolve(ret);
    }

    @PluginMethod
    public void remove(PluginCall call) {
        String key = call.getString("key", "");
        implementation.remove(key);
        call.resolve();
    }
}
